napari/layers/shapes/_accelerated_triangulate.py

```diff
@@ -0,0 +1,480 @@
+        vec_diff = path[i] - path[i - 1]
+        vec_len_arr[i - 1] = np.sqrt(vec_diff[0] ** 2 + vec_diff[1] ** 2)
+        normals[i - 1] = vec_diff / vec_len_arr[i - 1]
+
+    if closed:
+        vec_diff = path[0] - path[-1]
+        vec_len_arr[-1] = np.sqrt(vec_diff[0] ** 2 + vec_diff[1] ** 2)
+        normals[-1] = vec_diff / vec_len_arr[-1]
+
+    return normals, vec_len_arr * 0.5
+
+
+@njit(cache=True, inline="always")
+def _generate_2D_edge_meshes_loop(
+    path: np.ndarray,
+    closed: bool,
+    cos_limit: float,
+    bevel: bool,
+    normals: np.ndarray,
+    bevel_limit_array: np.ndarray,
+    centers: np.ndarray,
+    offsets: np.ndarray,
+    triangles: np.ndarray,
+) -> None:
+    if closed:
+        j = _set_centers_and_offsets(
+            centers,
+            offsets,
+            triangles,
+            path[0],
+            normals[-1],
+            normals[0],
+            bevel_limit_array[-1],
+            bevel_limit_array[0],
+            0,
+            cos_limit,
+            bevel,
+        )
+    else:
+        centers[0] = path[0]
+        centers[1] = path[0]
+        offsets[0, 0] = normals[0][1] * 0.5
+        offsets[0, 1] = -normals[0][0] * 0.5
+        offsets[1] = -offsets[0]
+        triangles[0] = [0, 1, 2]
+        triangles[1] = [1, 2, 3]
+        j = 2
+    for i in range(1, len(normals) - 1):
+        j += _set_centers_and_offsets(
+            centers,
+            offsets,
+            triangles,
+            path[i],
+            normals[i - 1],
+            normals[i],
+            bevel_limit_array[i - 1],
+            bevel_limit_array[i],
+            j,
+            cos_limit,
+            bevel,
+        )
+    if closed:
+        j += _set_centers_and_offsets(
+            centers,
+            offsets,
+            triangles,
+            path[-1],
+            normals[-2],
+            normals[-1],
+            bevel_limit_array[-2],
+            bevel_limit_array[-1],
+            j,
+            cos_limit,
+            bevel,
+        )
+        centers[j] = centers[0]
+        centers[j + 1] = centers[1]
+        offsets[j] = offsets[0]
+        offsets[j + 1] = offsets[1]
+    else:
+        centers[j] = path[-1]
+        centers[j + 1] = path[-1]
+        offsets[j, 0] = normals[-2][1] * 0.5
+        offsets[j, 1] = -normals[-2][0] * 0.5
+        offsets[j + 1] = -offsets[j]
+
+
+@njit(cache=True, inline="always")
+def _cut_end_if_repetition(path: np.ndarray) -> np.ndarray:
+    """Cut the last point of the path if it is the same as the second last point.
+
+    Parameters
+    ----------
+    path : np.ndarray
+        Nx2 array of central coordinates of path to be triangulated
+
+    Returns
+    -------
+    np.ndarray
+        Nx2 or (N-1)x2  array of central coordinates of path to be triangulated
+    """
+    path_ = np.asarray(path, dtype=np.float32)
+    if np.all(path_[-1] == path_[-2]):
+        return path_[:-1]
+    return path_
+
+
+@njit(cache=True)
```

Comment:
I have tried to remove this decorator, but it doubles execution time. 

With decorator:
![obraz](https://github.com/user-attachments/assets/79782822-c7ee-4dbc-93a3-9695fdafcfdd)

without a decorator:

![obraz](https://github.com/user-attachments/assets/74a8f5d2-d23a-40f2-9c6d-7f5b0a03b9c5)